[FEATURE] Add webmail links for gmail & yahoo mail

clevinson opened the pull request:
Closes https://github.com/teddywilson/defund12.org/issues/30 

Wasn't sure of the best way to do this style-wise, but this should add functionality for yahoo mail & gmail links.

I tried aol & outlook as well, but I think those URLs were error'ing out due to the email body's being too long.

teddywilson commented on _layouts/email.html:
```diff
@@ -6,8 +6,12 @@ <h2>{{ page.name }}</h2>
 		<div class='buttons'>
-			<a href="javascript:void(0);" onclick="openEmail()">Send email</a>
-			<a href="javascript:void(0);" onclick="copyToClipboard(this, `{{ page.permalink }}`, true)">Copy link</a>
+      <a href="javascript:void(0);" onclick="openEmail('mailto')">Send email</a>
+      <a href="javascript:void(0);" onclick="copyToClipboard(this, `{{ page.permalink }}`, true)">Copy link</a></br>
+			Send via webmail (
+        <a href="javascript:void(0);" onclick="openEmail('gmail')">gmail</a>
+        <a href="javascript:void(0);" onclick="openEmail('yahoo')">yahoo mail</a>
+      )
```
There seems to be a space before gmail and after yahoo mail (which should also be capitalized)

teddywilson replied:
see somment below about string literals

teddywilson commented on _layouts/email.html:
```diff
@@ -42,13 +46,20 @@ <h2>{{ page.name }}</h2>
+    const cc = `{{ page.cc | join: ',' }}`;
+    const ccText = cc !== null && cc.length > 0 ? `cc=${cc}&` : ``;
+
+    if (client === 'mailto') {
+      location.href = `mailto:${recipients}?${ccText}subject=${subject}&body=${body}`;
+    } else if (client === 'gmail') {
+      location.href = `https://mail.google.com/mail/?view=cm&fs=1&to=${recipients}&${ccText}su=${subject}&body=${body}`;
+    } else if (client === 'yahoo') {
+      location.href = `https://compose.mail.yahoo.com/?to=${recipients}&${ccText}subj=${subject}&body=${body}`;
+    }
```
these clients should be enumerated somewhere so they can be referenced from the function call as well. declaring as literals is dangerous.

teddywilson commented on _layouts/email.html:
```diff
@@ -57,7 +68,7 @@ <h2>{{ page.name }}</h2>
 			window.history.replaceState({}, document.title, `${window.location.origin}{{ page.permalink }}`)
 		} else {
-			openEmail()
+			openEmail('mailto')
```
ditto

teddywilson commented:
I left a few small comments, but I also think we need to discuss this with others on the team.

@minicreative, where do you weigh in here?

clevinson commented:
@teddywilson just addressed your comments above.

Personally I think the direct links like `defund12.org/nyc` should always default to what is currently "browse" mode (e.g. `defund12.org/nyc?browse`. This way it gives people the option of if they want a direct link, or to copy & paste certain texts.

Currently, when a person lands on the page, the `?browse` suffix dissapears quickly, and so if the person copies & pastes the link from the URL bar manually, they actually are copying a link to the mailto href, not the page they are viewing. This is confusing behavior IMO.

I opted for not really messing around with any of that functionality, and trying to just add in webmail link support without rocking the boat too much when it comes to how & when direct open links are triggered.

minicreative commented:
@clevinson Quick update - definitely want to follow up with this, we're prioritizing content for the time being. Folks on our team have a pretty strong consensus that the immediate mailto redirect is what makes our site effective. However, the client links are a great idea. I'm tagging this as a feature and we'll come back to it soon!
